src/luarocks/loader.lua

```diff
@@ -29,7 +29,7 @@ local require = nil
 
 -- Workaround for wrappers produced by older versions of LuaRocks
 local temporary_global = false
-if luarocks then
+if pcall(function() return luarocks end) then
```

Comment:
that is incorrect: `pcall` succeeds even though `luarocks` is nil. Need to test the second return value instead.

Reply:
I believe the updated patch does this, though it is still failing CI.

Reply:
Er, maybe I was looking at the outdated CI results. It seems to have passed now. I think this resolved, perhaps modulo variable name preferences.